### PR TITLE
[Yaml]Unify dygraph/static yaml definition for reshape

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR Category

Operator Mechanism

### PR Types

Deprecations

### Description

Pcard-67164

统一了reshape算子的动静yaml体系，为 https://github.com/PaddlePaddle/Paddle/pull/67009 的后续PR

### CODE REVIEW

File: paddle/phi/ops/yaml/inconsistent/static_ops.yaml [hunk 1]
```diff
@@ -708,19 +708,6 @@
...
-  output : Tensor(out)
-  infer_meta :
-    func : ReshapeInferMeta
-    spmd_rule : ReshapeInferSpmd
-  kernel :
-    func : reshape
-  inplace : (x -> out)
-  view: (x -> out)
-  backward: reshape_grad
-  interfaces : paddle::dialect::InferSymbolicShapeInterface
```
Comment (chen2016013) [inline paddle/phi/ops/yaml/inconsistent/static_ops.yaml:722]: 这里统一后没有 InferSymbolicShapeInterface，会有影响吗？

Comment (Aurelius84) [reply paddle/phi/ops/yaml/inconsistent/static_ops.yaml:722]: Thanks，已补充对齐

**Review by zyfncg — APPROVED** [review]

**Review by chen2016013 — APPROVED** [review]
LGTM

**Review by zhangbo9674 — APPROVED** [review]
LGTM

**Pull request merged**